### DISCUSSION TITLE
Are MacOS guests VMs supported?

### DISCUSSION THREAD

**Discussion by rastaman** [lima-vm/lima#2525] [Q&A] [answered]
Hi all,

I have searched a bit and didn't found any use of LimaVM to host MacOS VMs. I have an use case for this (a MacOS Jenkins build agent), and would like to know if it already exist and i have missed it and if it doesn't exist if it would be ok to add such support to LimaVM despite its name :-) If i am not wrong we use  https://github.com/Code-Hex/vz which support MacOS guests, so that should be only some plumbery to add the support? I would give it a try if it sounds ok and it doesn't exist.

Thanks in advance for your answers and your time,
Best regards

Edit: That doesn't look easy, there is lot of stuff to adapt to have support for all the features, but still i would be glad to be wrong :-)

**Comment by jandubois** [comment, marked as answer]
As you found out, it is not currently supported.

I also think it would be too much work to even try, but if you think you can make it work, by all means go for it. I would find it useful as well.

I would probably at other macOS VM projects to see if they can be adapted more easily. E.g. something like https://github.com/cirruslabs/tart

**Reply by rastaman** [reply]
Spot on @jandubois , thanks! Yes, tart fulfill all my needs in fact, and given all the work done on it i think it would be a waste of time to add all its configuration tweaks to Lima. Thanks for the hint!

**Reply by AkihiroSuda** [reply]
I don't think it would be a waste of time. PR will be welcomed, at least from my side.

**Reply by AkihiroSuda** [reply]
It should be also noted that tart is proprietary.

**Reply by jandubois** [reply]
> PR will be welcomed, at least from my side.

I said the same thing:

> if you think you can make it work, by all means go for it. I would find it useful as well.

I just think it would take more time than I would be willing to spend on it myself.

**Reply by afbjorklund** [reply]
Then again "mama" is a nice project name, no ? 🤡

**Comment by afbjorklund** [comment]
There was support for FreeBSD guests, which was a _little_ easier (with QEMU at least) but didn't have much of a use case...

* ~~https://github.com/lima-vm/lima/issues/1096~~
* https://github.com/lima-vm/lima/issues/1508

Even the read-only and non-standard variants like CoreOS and Container Linux are struggling to find their reason (in Lima).

* ~~https://github.com/lima-vm/lima/issues/1406~~
* https://github.com/lima-vm/lima/issues/1372

One of the issues was that neither supported `cloud-init`, so you had to resort to other means to do the basic configuration.

It also was bigger and slower, so it wasn't so useful as a base shared VM just to run containers on (like with WSL on Windows)

----

The container support on Mac is even worse, even if it does have some sandboxing and chrooting (like used in MacPorts).

And of course there are some licensing issues since it (macOS) is not Open Source anymore, and [Pure Darwin](https://www.puredarwin.org/) struggles...

But I think that the OS could be added, and use an external VM?

Then again, that's mostly a thin wrapper around `sshocker`.

**Reply by afbjorklund** [reply]
Then I remembered that (both) Mac's sshfs are proprietary.
